### PR TITLE
Small refactor for user search

### PR DESCRIPTION
**Pull request by fantasticfears**
In Rails spec, only `search_for_user` is extracted to helper. Changes are basically a place change.

### CODE REVIEW

File: app/assets/javascripts/discourse/lib/user-search.js.es6 [hunk 1]
```diff
@@ -44,7 +44,7 @@ function organizeResults(r, options) {
   if (r === CANCELLED_STATUS) { return r; }
 
   var exclude = options.exclude || [],
-      limit = options.limit || 5,
+      limit = options.limit || 6,
```
Comment (SamSaffron) [inline app/assets/javascripts/discourse/lib/user-search.js.es6]: this is just a bit off, the number of results are limited to 5... not 6. I think the change just creates confusion.

Comment (fantasticfears) [inline app/assets/javascripts/discourse/lib/user-search.js.es6]: Well, according to the previous logic, the intention is 6. I'll change that to 5.

**Comment by fantasticfears** [comment]
Spec checked and the test for limit is added.

**Comment by SamSaffron** [comment]
I am still somewhat confused... 

I thought what we do is search for 6, and render 5... The reason we do that is so we can conditionally show the "more results" thing. Not following what bug this behavior change is fixing.

**Comment by fantasticfears** [comment]
Ouch, I misunderstand your previous intention. I didn't alter the behaviour previously...

So my point is that as it iterates over the `result`, the default limit is closed range. However the array index starts at 0, this is so confusing when skim through the code as the limit reads 5 but returns 6 items. That costs me a hour to poke around in the test suite :( This is such a old practice problem since C.

Coming back to this part of code:
> I thought what we do is search for 6, and render 5... The reason we do that is so we can conditionally show the "more results" thing.

I am not sure where the code is explicit about searching for 6, Rails is default to 20. The limit value is used to trim the addition results.
image
I did a double check on Meta, it renders 6 users...

As a conclusion, `userSearch` tries to search for 20 users and shows 6 users by default...

**Comment by SamSaffron** [comment]
I have been struggling with this all day... but I am going to refer to: 

https://meta.discourse.org/t/discourse-development-contribution-guidelines/3823

In particular to the 

> Refactoring only NOT welcome

This changes the git history for the test file and does not really fix anything. I guess the test is in the wrong spot, but moving it should be up to a core developer. 

We have been bitten big times by refactoring PRs in the past, they are enormously difficult to deal with and the policy is there for a reason. 

Sorry.

**Comment by fantasticfears** [comment]
I see. I will work on the big fix pr instead.

**Comment by SamSaffron** [comment]
That is fine, just careful not to shuffle code around.

**Comment by fantasticfears** [comment]
Will do

**Pull request closed without merging**